resolver.go

```diff
@@ -73,8 +79,9 @@ func newResolver(ctx context.Context, cln *client.Client, p progress.Progress) *
 		}, p.Channel())
 		if err != nil {
 			r.done <- errors.Wrap(err, "resolver build failed")
+			return
 		}
-		close(r.done)
```

Comment:
this was causing a race on error where we might try to close `nil` on the resolver shutdown, fixed now to only modify r.done under lock on shutdown.